### DISCUSSION TITLE
The edge_index of one node

### DISCUSSION THREAD

**Discussion by aa-oo** [pyg-team/pytorch_geometric#6798] [Q&A]
Hello, I would like to ask, if there is only one node in my graph and I want to use gcn, how to define its edge_index? That's edge_index=torch.tensor([0],[0]],dtype=torch.long)? Thank you

**Comment by LukeLIN-web** [comment]
It is strange; how do you aggregate information from neighbors?

**Reply by aa-oo** [reply]
It doesn't have to consider its neighbors, in other words, the whole graph is equivalent to just one node, but I need to use gcn to convolve

**Reply by LukeLIN-web** [reply]
You can add self loop as `edge_index=torch.tensor([0],[0]],dtype=torch.long)`

**Comment by aa-oo** [comment]
ok,thank you!




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年2月27日(星期一) 中午1:29
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [pyg-team/pytorch_geometric] The edge_index of one node (Discussion #6798)





 
You can add self loop as edge_index=torch.tensor([0],[0]],dtype=torch.long)
